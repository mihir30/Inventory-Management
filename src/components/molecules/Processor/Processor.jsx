import React, { useState } from 'react';
import ArrowDropUpIcon from '@mui/icons-material/ArrowDropUp';
import DeleteIcon from '@mui/icons-material/Delete';
import ArrowDropDownIcon from '@mui/icons-material/ArrowDropDown';
import "./Processor.scss";
import SubProcessor from '../SubProcessor/SubProcessor';

const Processor = () => {
    const [open, setOpen] = useState(false);
    const [remove, setRemove] = useState(true);
    return (

        remove && <div className="processor-container">
            <div className='processor-header'>
                <div className='inner1' data-testid="inner-1">
                    {open ? <i className='processor-up-icon' onClick={() => setOpen(!open)}><ArrowDropUpIcon /></i> : <i className='processor-up-icon' onClick={() => setOpen(!open)}><ArrowDropDownIcon /></i>}
                    <span className='processor-label'>Processor</span>
                </div>
                <i className='delete-icon' onClick={() => setRemove(!remove)}><DeleteIcon /></i>
            </div>
            {open ?
                <div className='sub-group'>
                    <SubProcessor /></div> : <></>}
        </div>
    )
}


export default Processor;
