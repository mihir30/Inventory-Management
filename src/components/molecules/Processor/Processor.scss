.processor-container {
    display: block;
    margin-top: 0.5rem;
}
.processor-header {
    display: flex;
    background-color: gray;
    color: lightgray;
    align-items: center;
    padding: 0.4rem;
    justify-content: space-between;
}
.processor-header:hover {
    color: #4f4f51;
}
.delete-icon {
    float: right;
    cursor: pointer;
}
.inner1 {
    cursor: pointer;
}
.processor-content-input {
    margin-left: 1rem;
    border: none;
    height: 2rem;
    cursor: pointer;
}
.processor-content-div {
    display: flex;
    margin: 1rem;
    align-items: center;
    cursor: pointer;
}
.processor-content-label {
    font-size: small;
    font-weight: 500;
}
.sub-group .sub-input {
    border: none;
}
.sub-group .group2{
    display: flex;
    align-items: center;
}
.sub-group .star-mark{
    display: none;
}