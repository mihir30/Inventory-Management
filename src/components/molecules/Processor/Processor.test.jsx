import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import ReactDom from "react-dom";
import Processor from './Processor';

describe("Processor Component", () => {

    it('renders the component properly', () => {
        const divEl = document.createElement("div");
        ReactDom.render(<BrowserRouter><Processor></Processor></BrowserRouter>, divEl); 
        
    });

    it('should have the right processor component snapshot', () => {
        const tree = renderer.create().toJSON();
        expect(tree).toMatchSnapshot();
    })

});