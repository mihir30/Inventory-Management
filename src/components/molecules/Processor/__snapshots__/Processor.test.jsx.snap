// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Processor Component should have the right processor component snapshot 1`] = `null`;
